sdk/python/kfp/compiler/compiler.py

```diff
@@ -324,6 +324,20 @@ def _get_inputs_outputs_recursive_opsgroup(group):
 
     _get_inputs_outputs_recursive_opsgroup(root_group)
 
+    # Generate the input for SubGraph along with parallelfor
+    for sub_graph in opsgroup_groups:
+      if sub_graph in op_name_to_for_loop_op:
+        # get the parent of the current for-loop group
+        parent = opsgroup_groups[sub_graph][-2] 
```

Comment:
(consider this as a nitpick that doesn't block my sign off)

> `# get the parent of the current for-loop group`

This is kind of obvious from the code below. As I replied in the other thread, what's less obvious is why the parent is at index `-2`. Consider repeating or rephrasing the following comments, so that the readers of this code don't need to trace back to confirm the meaning of `-2`? You can decide.

https://github.com/kubeflow/pipelines/blob/8a22a89c7d075fe947d86377056a48c3fd0e7e5d/sdk/python/kfp/compiler/compiler.py#L104-L106

Reply:
I will update the comment.